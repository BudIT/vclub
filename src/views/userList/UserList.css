.menu {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  position: absolute;
  top: 61px;
  padding: 5px;
  right: 1px;
  min-width: 180px;
  background-color: #FFD00B;
  box-shadow: 0 2px 3px 0 rgba(0,0,0,0.4);
  overflow: hidden;
}

.icon {
  display: inline-flex;
  width: 30px;
  height: 30px;
  border-radius: 50%;
  box-shadow: 0 2px 3px 0 rgba(0,0,0,0.4);
  margin-bottom: 5px;
  margin-right: 5px;
  background: darkorange;
}

.user {
  text-align: center;
  align-self: center;
  width: 100%;
}

.master {
  font-weight: bold;
}
