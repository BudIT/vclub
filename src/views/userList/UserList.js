import React, { PropTypes } from 'react';
import Transition from 'react-motion-ui-pack';
import style from './UserList.css';

function UserList(props) {
  const {
    members,
  } = props;

  const sortedMembers = members.sort((a, b) => {
    if (a.master < b.master) {
      return 1;
    }
    if (a.master > b.master) {
      return -1;
    }
    return 0;
  });

  return (
    <Transition
      className={style.menu}
      component="div"
      enter={{
        opacity: 1,
      }}
      leave={{
        height: 0,
        opacity: 0,
      }}
    >
      {sortedMembers.map(member => (
        <span key={member.id}>
          <span className={style.icon}>
            <i className={style.user}>{member.name.charAt(0)}</i>
          </span>
          {member.name}
        </span>
        ))}
    </Transition>
  );
}

UserList.propTypes = {
  members: PropTypes.arrayOf(React.PropTypes.object).isRequired,
};

export default UserList;
