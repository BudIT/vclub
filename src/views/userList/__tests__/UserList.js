/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
/* eslint-env jest */

import React from 'react';
import { mount } from 'enzyme';
import UserList from '../UserList';

const usersWithFewMasters = [
  { id: 0, name: 'Misato', master: false },
  { id: 1, name: 'Simon', master: false },
  { id: 2, name: 'Jack London', master: true },
  { id: 3, name: 'Master', master: true },
  { id: 4, name: 'And another master', master: true },
  { id: 5, name: 'Not master', master: false },
  { id: 6, name: 'Harry West', master: false },
];

test('<UserList /> renders correctly', () => {
  const rendered = mount(
    <UserList members={usersWithFewMasters} />
  );
  expect(rendered.html()).toMatchSnapshot();
});
