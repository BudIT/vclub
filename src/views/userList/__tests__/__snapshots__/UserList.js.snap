exports[`test <UserList /> renders correctly 1`] = `"<div class=\"menu\"><span style=\"height: 0px; opacity: 0;\"><span class=\"icon\"><i class=\"user\">J</i></span><b class=\"master\">Jack London</b><div class=\"erd_scroll_detection_container erd_scroll_detection_container_animation_active\" style=\"visibility: hidden; display: inline; width: 0px; height: 0px; z-index: -1; overflow: hidden; margin: 0px; padding: 0px;\"></div></span><span style=\"height: 0px; opacity: 0;\"><span class=\"icon\"><i class=\"user\">M</i></span><b class=\"master\">Master</b><div class=\"erd_scroll_detection_container erd_scroll_detection_container_animation_active\" style=\"visibility: hidden; display: inline; width: 0px; height: 0px; z-index: -1; overflow: hidden; margin: 0px; padding: 0px;\"></div></span><span style=\"height: 0px; opacity: 0;\"><span class=\"icon\"><i class=\"user\">A</i></span><b class=\"master\">And another master</b><div class=\"erd_scroll_detection_container erd_scroll_detection_container_animation_active\" style=\"visibility: hidden; display: inline; width: 0px; height: 0px; z-index: -1; overflow: hidden; margin: 0px; padding: 0px;\"></div></span><span style=\"height: 0px; opacity: 0;\"><span class=\"icon\"><i class=\"user\">M</i></span><!-- react-text: 17 -->Misato<!-- /react-text --><div class=\"erd_scroll_detection_container erd_scroll_detection_container_animation_active\" style=\"visibility: hidden; display: inline; width: 0px; height: 0px; z-index: -1; overflow: hidden; margin: 0px; padding: 0px;\"></div></span><span style=\"height: 0px; opacity: 0;\"><span class=\"icon\"><i class=\"user\">S</i></span><!-- react-text: 21 -->Simon<!-- /react-text --><div class=\"erd_scroll_detection_container erd_scroll_detection_container_animation_active\" style=\"visibility: hidden; display: inline; width: 0px; height: 0px; z-index: -1; overflow: hidden; margin: 0px; padding: 0px;\"></div></span><span style=\"height: 0px; opacity: 0;\"><span class=\"icon\"><i class=\"user\">N</i></span><!-- react-text: 25 -->Not master<!-- /react-text --><div class=\"erd_scroll_detection_container erd_scroll_detection_container_animation_active\" style=\"visibility: hidden; display: inline; width: 0px; height: 0px; z-index: -1; overflow: hidden; margin: 0px; padding: 0px;\"></div></span><span style=\"height: 0px; opacity: 0;\"><span class=\"icon\"><i class=\"user\">H</i></span><!-- react-text: 29 -->Harry West<!-- /react-text --><div class=\"erd_scroll_detection_container erd_scroll_detection_container_animation_active\" style=\"visibility: hidden; display: inline; width: 0px; height: 0px; z-index: -1; overflow: hidden; margin: 0px; padding: 0px;\"></div></span></div>"`;
